wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -0,0 +1,892 @@
+            security_api.patch_security_context_constraints(name=mapping['scc'],
+                                                            body={'users': got_users})
+
+        # grant roles to orchestrator
+        self.logger.info("assigning additional roles to cfme-orchestrator")
+        auth_api = self.ociclient.AuthorizationOpenshiftIoV1Api()
+        orchestrator_sa = self.kclient.V1ObjectReference(name='cfme-orchestrator',
+                                                         kind='ServiceAccount',
+                                                         namespace=proj_name)
+
+        view_role = self.kclient.V1ObjectReference(name='view')
+        view_role_binding_name = self.kclient.V1ObjectMeta(name='view')
+        view_role_binding = self.ociclient.V1RoleBinding(role_ref=view_role,
+                                                         subjects=[orchestrator_sa],
+                                                         metadata=view_role_binding_name)
+        self.logger.debug("creating 'view' role binding "
+                          "for cfme-orchestrator sa in project %s", proj_name)
+        auth_api.create_namespaced_role_binding(namespace=proj_name, body=view_role_binding)
+
+        edit_role = self.kclient.V1ObjectReference(name='edit')
+        edit_role_binding_name = self.kclient.V1ObjectMeta(name='edit')
+        edit_role_binding = self.ociclient.V1RoleBinding(role_ref=edit_role,
+                                                         subjects=[orchestrator_sa],
+                                                         metadata=edit_role_binding_name)
+        self.logger.debug("creating 'edit' role binding "
+                          "for cfme-orchestrator sa in project %s", proj_name)
+        auth_api.create_namespaced_role_binding(namespace=proj_name, body=edit_role_binding)
+
+        self.logger.info("project sa created via api have no some mandatory roles. adding them")
+        self._restore_missing_project_role_bindings(namespace=proj_name)
+
+        # creating pods and etc
+        processing_params = {'DATABASE_PASSWORD': password,
+                             'APPLICATION_DOMAIN': proj_url}
+        processing_params.update(prepared_tags)
+        self.logger.info(("processing template and passed params in order to "
+                          "prepare list of required project entities"))
+        template_entities = self.process_template(name=template, namespace=self.default_namespace,
+                                                  parameters=processing_params)
+        self.logger.debug("template entities:\n {e}".format(e=template_entities))
+        kinds = set([e['kind'] for e in template_entities])
+        entity_names = {e: inflection.underscore(e) for e in kinds}
+        proc_names = {k: 'create_{e}'.format(e=p) for k, p in entity_names.items()}
+        for entity in template_entities:
+            if entity['kind'] in kinds:
```

Comment:
I don't think this can be false, since `kinds` was generated from the `kind` attributes in `template_entities` on L249. Just checking in case in order to error log?

Reply:
I implemented create methods only for entities present in template right now.
However, it is possible that template will contain entities of another type in future.